### PR TITLE
Adding configurable maximum subnet variable and more descriptive tagging

### PR DESCRIPTION
**Pull request by dcowan-vestmark**
## what
*  Allows the ability to override the max number of subnets to create. 

## why 
For some regions with many availability zones the amount of subnets gets extreme.  This allows the ability to override the max number of subnets to create.  Still defaults to all if nothing is supplied.

### CODE REVIEW

**Comment by dcowan-vestmark** [comment]
@osterman please review

File: nat.tf [hunk 1]
```diff
@@ -1,10 +1,21 @@
+module "nat_label" {
+  source     = "git::https://github.com/cloudposse/terraform-null-label.git?ref=tags/0.2.2"
```
Comment (aknysh) [inline nat.tf]: use the latest version here
`tags/0.3.3`

File: nat.tf [hunk 2]
```diff
@@ -1,10 +1,21 @@
...
+}
+
 locals {
   nat_gateways_count = "${var.nat_gateway_enabled == "true" ? length(var.availability_zones) : 0}"
 }
 
 resource "aws_eip" "default" {
   count = "${local.nat_gateways_count}"
   vpc   = true
+  tags  = "${module.private_label.tags}"
```
Comment (aknysh) [inline nat.tf]: did you mean using `nat_label` here

File: nat.tf [hunk 3]
```diff
@@ -15,6 +26,7 @@ resource "aws_nat_gateway" "default" {
   count         = "${local.nat_gateways_count}"
   allocation_id = "${element(aws_eip.default.*.id, count.index)}"
   subnet_id     = "${element(aws_subnet.public.*.id, count.index)}"
+  tags          = "${module.private_label.tags}"
```
Comment (aknysh) [inline nat.tf]: did you mean using `nat_label` here

File: private.tf [hunk 4]
```diff
@@ -12,14 +12,18 @@ module "private_subnet_label" {
   source    = "git::https://github.com/cloudposse/terraform-null-label.git?ref=tags/0.3.3"
   namespace = "${var.namespace}"
   stage     = "${var.stage}"
-  name      = "private"
+  name      = "${var.name}-private"
```
Comment (aknysh) [inline private.tf]: I'm torn here.
On the one hand, we need to follow the pattern and add `name` to every resource in a project.
In this case, it's better to use `attributes` and not hardcode name prefixes:

```
  name       = "${var.name}"
  attributes = ["private"]
```

but on the other hand, the subnets are for the entire infrastructure and many different applications could be placed into the subnets (e.g. app servers, Postgres, Redis, etc.), all with their own different names.
So here we are talking about just the names of the subnets itself.
If we follow the same pattern, then we could do this
```
  name       = "subnet"
  attributes = ["private"]
```
But we do not use the resource types in resource names.
That's why we just named the subnets `private` and `public`.

**Comment by osterman** [comment]
Can you attach the output of a `terraform plan`? I think that will help visualize the changes.

File: private.tf [hunk 5]
```diff
@@ -28,6 +32,10 @@ resource "aws_subnet" "private" {
   }
 }
 
+output blahpriv {
```
[the remark this answers is not in the document]
Comment (dscowan) [reply private.tf]: This was just to help me debug, I will remove

**Comment by dscowan** [comment]
@osterman  here is the plan

[plan-out.txt](https://github.com/cloudposse/terraform-aws-dynamic-subnets/files/1964117/plan-out.txt)

File: nat.tf (hunk 3 above)
Comment (dscowan) [reply nat.tf]: Shouldn't they be the same?

**Comment by dscowan** [comment]
@osterman @aknysh  All comments should be addressed now

File: nat.tf (hunk 2 above)
Comment (osterman) [reply nat.tf]: @dcowan-vestmark did you see this comment?

File: public.tf [hunk 6]
```diff
@@ -2,7 +2,7 @@ module "public_subnet_label" {
   source    = "git::https://github.com/cloudposse/terraform-null-label.git?ref=tags/0.3.3"
   namespace = "${var.namespace}"
   stage     = "${var.stage}"
-  name      = "public"
+  name      = "${var.name}-public"
```
Comment (osterman) [inline public.tf]: Use attributes:  (like here)

https://github.com/cloudposse/terraform-aws-dynamic-subnets/pull/29/files#diff-67814bdfab2d6469a77bb872064fde02R16

File: variables.tf [hunk 7]
```diff
@@ -36,6 +36,10 @@ variable "region" {
   description = "AWS Region (e.g. `us-east-1`)"
 }
 
+variable "max_subnet_count" {
```
Comment (osterman) [inline variables.tf]: Add a `description` to this variable. Explain what `0` means.

**Comment by aknysh** [comment]
@dcowan-vestmark 
can you address the comments above so we can do final review and merge
thanks a lot!!

**Comment by osterman** [comment]
Thanks @dcowan-vestmark !

**Pull request merged**